systemtest/src/test/java/io/strimzi/systemtest/AbstractClusterTest.java

```diff
@@ -152,4 +154,36 @@ String globalVariableJsonPathBuilder(String variable) {
                 .filter(event -> event.getInvolvedObject().getName().equals(resourceName))
                 .collect(Collectors.toList());
     }
+
+    public void sendMessages(List<String> messages, String clusterName, String topic) {
+        LOGGER.info("Sending messages");
+        String messagesToSend = "";
+        for (int i = 0; i < messages.size(); i++) {
+            messagesToSend = messagesToSend + messages.get(i) + "\\n";
+        }
+        messagesToSend = messagesToSend.substring(0, messagesToSend.length() - 2);
```

Comment:
what is a reason for using `messagesToSend.length() - 2`?

Reply:
I need to remove last '\n' characters. Otherwise, producer will generate an extra empty message.

Reply:
I've not yet looked at this PR in detail, but if all you're doing here is joining `messages` with `\n` then `final String messagesToSend =messages.stream().collect(Collectors.joining("\n"));` should be all you need. Or there are plenty of utility libraries which provide a `join` function.

Concatenating to a `messagesToSend` String in a loop is the wrong way to do it because each iteration of the loop allocates temporary String, which then needs to be GC'd. Using a `StringBuilder` is better (being mutable means less allocation and GC, but still there'll be some).